metals/src/main/scala/scala/meta/internal/metals/codeactions/AddingBracesCodeAction.scala

```diff
@@ -0,0 +1,641 @@
+   * @param document
+   * @param blockEmbraceable the branch of the main tree which can potentially have
+   *                         the type `Term.Block`; so can be braced.
+   * @param codeActionSubjectTitle
+   * @return
+   */
+  private def createCodeActionForPotentialBlockHolder(
+      blockHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      blockEmbraceable: Term,
+      codeActionSubjectTitle: String,
+      bracelessStart: String = ""
+  ): Option[l.CodeAction] = {
+    val indentation =
+      getIndentationForPositionInDocument(blockHolder.pos, document)
+
+    if (!isBlockEmbraceableBraced(blockEmbraceable)) {
+      createCodeActionToTakePotentialBlockHolderBraceful(
+        blockHolder,
+        path,
+        document,
+        blockEmbraceable,
+        codeActionSubjectTitle,
+        indentation,
+        bracelessStart
+      )
+    } else None
+  }
+
+  private def maybeGetEndMarkerPos(tree: Tree): Option[Position] =
+    tree.parent
+      .flatMap(parent =>
+        parent.children.dropWhile(_ != tree).tail.headOption.collectFirst {
+          case endMarker: Term.EndMarker => endMarker
+        }
+      )
+      .map(_.pos)
+
+  private def createCodeActionForTemplateHolder(
+      templateHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      templ: Template,
+      codeActionSubjectTitle: String
+  ): Option[l.CodeAction] = {
+    val indentation =
+      getIndentationForPositionInDocument(templateHolder.pos, document)
+    val expectedBraceStartPos = util
+      .Try {
+        val lastInit = templ.inits.maxBy(init => init.pos.end)
+        lastInit.pos.end
+      }
+      .getOrElse(templ.pos.start)
+    if (!hasBraces(templ)) {
+      createCodeActionToTakeTemplateHolderBraceful(
+        templateHolder,
+        path,
+        document,
+        templ,
+        codeActionSubjectTitle,
+        expectedBraceStartPos,
+        indentation
+      )
+    } else None
+  }
+
+  private def createCodeActionToTakeTemplateHolderBraceful(
+      templateHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      templ: Template,
+      codeActionSubjectTitle: String,
+      expectedBraceStartPos: Int,
+      indentation: String
+  ): Option[l.CodeAction] = {
+
+    for {
+      colonPose <- templ.tokens
+        .find(token =>
+          token.text == ":" && token.pos.start >= expectedBraceStartPos
+        )
+        .map(_.pos.toLSP.getStart)
+    } yield createCodeActionForGoingBraceful(
+      path,
+      expectedBraceStartPos = colonPose,
+      expectedBraceEndPose = templ.pos.toLSP.getEnd,
+      bracelessStart = ":",
+      bracelessEnd = "",
+      indentation = indentation,
+      document = document,
+      codeActionSubjectTitle = codeActionSubjectTitle,
+      maybeEndMarkerPos = maybeGetEndMarkerPos(templateHolder)
+    )
+  }
+
+  private def createCodeActionForGoingBraceful(
+      path: AbsolutePath,
+      indentation: String,
+      document: String,
```

Comment:
Seems unused

Reply:
This is the common part of code between all functions which add braces. So it is refactored here. It is called in 
`createCodeActionToTakePotentialBlockHolderBraceful`
and `createCodeActionToTakeTemplateHolderBraceful`

Reply:
But it's not needed, is it? I don't see it used in this function.

Also I think you can shorter the names a bit like:
`blockBracefulAction` and `templateBracelessAction`

otherwise it's hard to read.